### PR TITLE
ppOpen-APPL/FEM: New package.

### PR DESCRIPTION
**Pull request by tkameyama**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ppopen-appl-fem/package.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    homepage = "http://ppopenhpc.cc.u-tokyo.ac.jp/ppopenhpc/"
+    url = "file://{0}/ppohFEM_1.0.1.tar.gz".format(os.getcwd())
+
+    version('1.0.1', sha256='eea8837fa3eda284759b7ebf27c27cea8cbf9cf65cf37c62941700e1321aeb07')
+
+    depends_on('mpi')
+    depends_on('metis')
+
+    # gcc dose not support OpenMP atomic to same structure refernce.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ppopen-appl-fem/package.py:26]:
```suggestion
    # gcc does not support OpenMP atomic to same structure reference.
```

File: var/spack/repos/builtin/packages/ppopen-appl-fem/package.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            r'^F90OPTFLAGS *=.*$',
+            'F90OPTFLAGS = {0}'.format(' '.join(fflags))
+        )
+        makefile_in.filter(
+            r'^METISDIR *=.*$',
+            'METISDIR = {0}'.format(spec['metis'].prefix)
+        )
+        makefile_in.filter('mpicc', spec['mpi'].mpicc)
+        makefile_in.filter('mpif90', spec['mpi'].mpifc)
+        makefile_in.filter('-openmp', self.compiler.openmp_flag)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ppopen-appl-fem/package.py:54]: Some of these changes seem contradictory. The patch above replaces `-openmp` with `-fopenmp`, then here you try to replace `-openmp` with `self.compiler.openmp_flag`. Maybe you should only patch it once (preferably here where you can make it work for any compiler).

File: var/spack/repos/builtin/packages/ppopen-appl-fem/package.py (hunk 1 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/ppopen-appl-fem/package.py]: fixed

File: var/spack/repos/builtin/packages/ppopen-appl-fem/package.py (hunk 2 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/ppopen-appl-fem/package.py]: fixed.
Thank you.

**Pull request merged**